Mathlib/NumberTheory/Cyclotomic/Rat.lean

```diff
@@ -331,16 +332,105 @@ theorem zeta_sub_one_prime [IsCyclotomicExtension {p ^ (k + 1)} ℚ K]
 /-- `ζ - 1` is prime if `ζ` is a primitive `p`-th root of unity. -/
 theorem zeta_sub_one_prime' [h : IsCyclotomicExtension {p} ℚ K] (hζ : IsPrimitiveRoot ζ p) :
     Prime ((hζ.toInteger - 1)) := by
-  convert zeta_sub_one_prime (k := 0) (by simpa)
-  simpa
+  convert zeta_sub_one_prime (k := 0) (by simpa only [zero_add, pow_one])
+  simpa only [zero_add, pow_one]
 
 theorem subOneIntegralPowerBasis_gen_prime [IsCyclotomicExtension {p ^ (k + 1)} ℚ K]
     (hζ : IsPrimitiveRoot ζ ↑(p ^ (k + 1))) :
-    Prime hζ.subOneIntegralPowerBasis.gen := by simpa using hζ.zeta_sub_one_prime
+    Prime hζ.subOneIntegralPowerBasis.gen := by simpa only [subOneIntegralPowerBasis_gen] using
+      hζ.zeta_sub_one_prime
 
 theorem subOneIntegralPowerBasis'_gen_prime [IsCyclotomicExtension {p} ℚ K]
     (hζ : IsPrimitiveRoot ζ ↑p) :
-    Prime hζ.subOneIntegralPowerBasis'.gen := by simpa using hζ.zeta_sub_one_prime'
+    Prime hζ.subOneIntegralPowerBasis'.gen := by simpa only [subOneIntegralPowerBasis'_gen] using
+      hζ.zeta_sub_one_prime'
+
+open nonZeroDivisors in
+/-- The norm, relative to `ℤ`, of `ζ ^ p ^ s - 1` in a `p ^ (k + 1)`-th cyclotomic extension of `ℚ`
+is p ^ p ^ s` if `s ≤ k` and `p ^ (k - s + 1) ≠ 2`. -/
+lemma norm_toInteger_pow_sub_one_of_prime_pow_ne_two [IsCyclotomicExtension {p ^ (k + 1)} ℚ K]
+    (hζ : IsPrimitiveRoot ζ ↑(p ^ (k + 1))) {s : ℕ} (hs : s ≤ k) (htwo : p ^ (k - s + 1) ≠ 2) :
+    Algebra.norm ℤ (hζ.toInteger ^ (p : ℕ) ^ s - 1) = p ^ (p : ℕ) ^ s := by
+  apply (algebraMap ℤ ℚ).injective_int
+  have : NumberField K := IsCyclotomicExtension.numberField {p ^ (k + 1)} ℚ K
+  have : algebraMap (𝓞 K) K (hζ.toInteger ^ (p : ℕ) ^ s - 1) = ζ ^ (p : ℕ) ^ s - 1 := by
+    simp only [map_sub, map_pow, map_one, sub_left_inj]
+    exact rfl
+  rw [← Algebra.norm_localization (Sₘ := K) ℤ ℤ⁰, this, hζ.norm_pow_sub_one_of_prime_pow_ne_two
+    (cyclotomic.irreducible_rat
+    (by simp only [PNat.pow_coe, gt_iff_lt, PNat.pos, pow_pos])) hs htwo]
```

Comment:
This seems a little clearer to me; alternatively you could move the `norm_pow_sub_one_of_prime_pow_ne_two` call into a separate `have` as well
```suggestion
  rw [← Algebra.norm_localization (Sₘ := K) ℤ ℤ⁰, this,
    hζ.norm_pow_sub_one_of_prime_pow_ne_two
      (cyclotomic.irreducible_rat (by simp only [PNat.pow_coe, gt_iff_lt, PNat.pos, pow_pos]))
      hs htwo]
```